mikeio1d/result_network/result_catchments.py

```diff
@@ -50,3 +59,20 @@ def set_res1d_catchment_to_dict(self, catchment):
         Create a dict entry from catchment ID to ResultCatchment object.
         """
         self[catchment.id] = catchment
+
+    def to_geopandas(self) -> GeoDataFrame:
+        """
+        Convert catchments to a geopandas.GeoDataFrame object.
+
+        Returns
+        -------
+        gdf : geopandas.GeoDataFrame
+            A GeoDataFrame object with catchments as Polygon geometries.
+        """
+        gpd = try_import_geopandas()
+        ids = [catchment.id for catchment in self.values()]
+        geometries = [catchment.geometry.to_shapely() for catchment in self.values()]
```

Comment:
Introduce a variable catchments = self.values() and use it throughout the method.